server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/CassandraSessionModule.java

```diff
@@ -79,19 +83,24 @@ CassandraSessionConfiguration getCassandraSessionConfiguration(FileSystem fileSy
     @Singleton
     Cluster provideCluster(CassandraSessionConfiguration cassandraSessionConfiguration, AsyncRetryExecutor executor) throws FileNotFoundException, ConfigurationException, ExecutionException, InterruptedException {
         PropertiesConfiguration configuration = cassandraSessionConfiguration.getConfiguration();
-
+        List<CassandraServer> servers = listCassandraServers(configuration);
         return getRetryer(executor, configuration)
                 .getWithRetry(ctx -> ClusterWithKeyspaceCreatedFactory
                         .config(
-                            ClusterFactory.createClusterForSingleServerWithoutPassWord(
-                                configuration.getString("cassandra.ip"),
-                                configuration.getInt("cassandra.port")),
-                                configuration.getString("cassandra.keyspace"))
+                            ClusterFactory.createClusterForClusterWithoutPassWord(servers),
+                            configuration.getString("cassandra.keyspace"))
                         .replicationFactor(configuration.getInt("cassandra.replication.factor"))
                         .clusterWithInitializedKeyspace())
                 .get();
     }
 
+    private List<CassandraServer> listCassandraServers(PropertiesConfiguration configuration) {
+        String[] ips = configuration.getStringArray("cassandra.ip");
+        int port = configuration.getInt("cassandra.port");
+        List<CassandraServer> servers = Stream.of(ips).map(ip -> new CassandraServer(ip, port)).collect(Guavate.toImmutableList());
```

Comment:
And you can return without assigning servers variable. As the name is alrady conveyed by the method name.